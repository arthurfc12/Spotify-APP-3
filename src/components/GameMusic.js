import React from 'react';
import './App.css';
import SelectMusic from './SelectMusic.js';

class GameMusic extends React.Component {

    constructor(props){
        super(props)
        this.state={
            loading: false,
            playlists: "",
            tracks: "",
        }
    }

    componentWillMount() {
        
        const requestOptions = {
            method: "GET",
            headers: {
                "Authorization": "Bearer " + this.props.token
            }
          }
          
          fetch('https://api.spotify.com/v1/me/playlists', requestOptions)
              .then(response => response.json())
              .then(data => this.setState({playlists: data.items 
          }))

    }


    render(){
        if(this.state.playlists !== ""){
            return(
                <div>
                    <h1 className="titulos">Game Acerte a MÃºsica</h1>
                    {
                        this.state.loading ? <h1>Carregando</h1> :
                        <div>
                            <SelectMusic playlistList = {this.state.playlists} token = {this.props.token}/>
                        </div>
                    }
                </div>
            );
        }
        else {
            return(
                <h1 className="titulos">Carregando</h1>
            );
        }
    }
}
export default GameMusic;