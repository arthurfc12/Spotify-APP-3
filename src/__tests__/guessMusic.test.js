import React from 'react';
import { render, wait, cleanup, waitFor } from '@testing-library/react';
import mockedFetch from 'fetch';
import toJson from 'enzyme-to-json';
import GameMusic from '../components/GameMusic';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

afterEach(cleanup);

it('should appear the name of the correct Music', async () => {
    const data = {
      props: {
        album: [
          {
            name: 'Who Says',
            artists: [
                {name: 'Selena Gomez & The Scene'}
            ],
          },
        ],
      },
    };
  
    mockedFetch.get.mockResolvedValueOnce(data);
    const app = shallow(<GameMusic />);
  
    expect(toJson(app)).toMatchSnapshot();
    // await waitFor(() => {
    // });
  });
  