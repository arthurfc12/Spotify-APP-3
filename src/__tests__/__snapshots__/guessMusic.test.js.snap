// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should appear the name of the correct Music 1`] = `
<h1
  className="titulos"
>
  Carregando
</h1>
`;
